app/controllers/vendor_sweets_controller.rb

```diff
@@ -0,0 +1,18 @@
+class VendorSweetsController < ApplicationController
+  def create
+    @vendor = Vendor.find(vendor_sweet_params[:vendor_id])
+    @vendor_sweet = @vendor.vendor_sweets.build(vendor_sweet_params)
+
+    if @vendor_sweet.save
+      redirect_to vendor_path(@vendor)
+    else
+      render 'vendors/show'
```

Comment:
Be consistent with the quotes you're using. Try to always use double quotes:
```suggestion
      render "vendors/show"
```